### PR TITLE
Add support for hpu_backend and Resnet50 compile example

### PR DESCRIPTION
**Pull request by wozna**
## Description

The changes add support for hpu_backend and provide an example for resnet50 for compile. 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Feature/Issue validation/testing

In order to show how to run torchserve on HPU, I added an example of running Resnet50 with a description and a modified handler. 

## Checklist:

- [x] Did you have fun?
- [x] Have you added tests that prove your fix is effective or that this feature works?
- [x] Has code been commented, particularly in hard-to-understand areas?
- [x] Have you made corresponding changes to the documentation?

### CODE REVIEW

**Comment by wozna** [comment]
@agunapal Can I ask you to review my proposal?

**Comment by agunapal** [comment]
> @agunapal Can I ask you to review my proposal?

Hi @wozna Thanks for the PR. Please wait till https://github.com/pytorch/serve/pull/3166 is merged. You would have to edit your PR once this is merged.

**Review by agunapal — CHANGES_REQUESTED** [review]

File: examples/pt2/torch_compile_hpu/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+```
+`pt2: {backend: "hpu_backend"}` - this line enables compile mode, if you remove it from the config file, the model will run in eager mode.
+
+### 2. Create model archive
+
+Download the pre-trained model and prepare the model archive:
+```bash
+wget https://download.pytorch.org/models/resnet50-11ad3fa6.pth
+mkdir model_store
+PT_HPU_LAZY_MODE=0 torch-model-archiver --model-name resnet-50 --version 1.0 --model-file model.py \
```
[the remark this answers is not in the document]
Comment (wozna) [reply examples/pt2/torch_compile_hpu/README.md:50]: Sure, Gaudi integration with PyTorch supports officially 2 modes of operation: lazy and eager+torch.compile (beta state). Currently the first one is default, but we plan to move to eager+torch.compile. PT_HPU_LAZY_MODE=0 selects eager+torch.compile mode and we will remove this flag as we switch to eager+torch.compile as default solution (1-2 releases) 
Do you think I should add this explanation to readme.md as well?

[the remark this answers is not in the document]
Comment (agunapal) [reply examples/pt2/torch_compile_hpu/README.md:50]: Yes, please add that in the Readme, you can add links as well

File: examples/pt2/torch_compile_hpu/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  "lens_cap": 0.002257900545373559
+}
+```
+
+### 5. Stop the server
+Stop TorchServe with the following command:
+
+```bash
+torchserve --stop
+```
```
[the remark this answers is not in the document]
Comment (agunapal) [reply examples/pt2/torch_compile_hpu/README.md:95]: Hi @wozna If you can please update this, we can merge this PR

[the remark this answers is not in the document]
Comment (RafLit) [reply examples/pt2/torch_compile_hpu/README.md:95]: Hi @agunapal Sorry for the delay. The config has been updated to show the perf gain.

File: examples/pt2/torch_compile_hpu/README.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+`torch.compile` needs a few inferences to warmup. Once warmed up, we see the following
+```bash
+[INFO ] W-9000-resnet-50_1.0-stdout org.pytorch.serve.wlm.WorkerLifeCycle - result=[METRICS]ts_handler_preprocess.Milliseconds:6.833314895629883|#ModelName:resnet-50,Level:Model|#type:GAUGE|###,1718265582,53da9032-4ad3-49df-8cd4-2d499eea7691, pattern=[METRICS]
+[INFO ] W-9000-resnet-50_1.0-stdout org.pytorch.serve.wlm.WorkerLifeCycle - result=[METRICS]ts_handler_inference.Milliseconds:0.7846355438232422|#ModelName:resnet-50,Level:Model|#type:GAUGE|###,1718265582,53da9032-4ad3-49df-8cd4-2d499eea7691, pattern=[METRICS]
+[INFO ] W-9000-resnet-50_1.0-stdout org.pytorch.serve.wlm.WorkerLifeCycle - result=[METRICS]ts_handler_postprocess.Milliseconds:1.9681453704833984|#ModelName:resnet-50,Level:Model|#type:GAUGE|###,1718265582,53da9032-4ad3-49df-8cd4-2d499eea7691, pattern=[METRICS]
+```
+
+### Conclusion
+
+`torch.compile` reduces the inference time from 5.22ms to 0.78ms
```
Comment (agunapal) [inline examples/pt2/torch_compile_hpu/README.md:142]: Impressive gain!

File: examples/pt2/torch_compile_hpu/README.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
+
+# TorchServe Inference with torch.compile with HPU backend of Resnet50 model
+
+This guide provides steps on how to optimize a ResNet50 model using `torch.compile` with [HPU backend](https://docs.habana.ai/en/latest/PyTorch/Inference_on_PyTorch/Getting_Started_with_Inference.html), aiming to enhance inference performance when deployed through TorchServe. `torch.compile` allows for ahead-of-time compilation of PyTorch models.
```
Comment (agunapal) [inline examples/pt2/torch_compile_hpu/README.md:4]: `torch.compile` is a JIT compiler. Can you please change this. To help with warmup problem, we use AOT compile @RafLit

**Review by agunapal — CHANGES_REQUESTED** [review]

File: examples/pt2/torch_compile_hpu/README.md (hunk 4 above)
Comment (agunapal) [reply examples/pt2/torch_compile_hpu/README.md:4]: something like this : `torch.compile` allows for JIT compilation of Python code into optimized kernels with a simple API.

Comment (RafLit) [reply examples/pt2/torch_compile_hpu/README.md]: The description has been updated according to your suggestion @agunapal

**Review by agunapal — APPROVED** [review]
LGTM
